src/einsteinpy/bodies.py

```diff
@@ -15,195 +15,192 @@
+from textwrap import dedent
+
 from astropy import units as u
 
-from einsteinpy import constant
+from einsteinpy.constant import R_sun, Solar_Mass
 
 __all__ = ["Body"]
 
 
+@dataclass
 class Body:
     """
     Class to create a generic Body
+
+    Parameters
+    ----------
+    name : string
+        Name or ID of the body
+    mass : ~astropy.units.kg
+        Mass of the body
+    q : ~astropy.units.C, optional
+        Charge on the body
+    R : ~astropy.units
+        Radius of the body
+    differential : ~einsteinpy.coordinates.differential.*, optional
+        Complete coordinates of the body
+    parent : Body, optional
+        The parent object of the body
+        Useful in case of multibody systems
     """
 
-    @u.quantity_input(mass=u.kg, R=u.km)
-    def __init__(
-        self,
-        name="Generic Body",
-        mass=0 * u.kg,
-        q=0 * u.C,
-        R=0 * u.km,
-        differential=None,
-        parent=None,
-    ):
-        """
-        Parameters
-        ----------
-        name : string
-            Name or ID of the body
-        mass : ~astropy.units.kg
-            Mass of the body
-        q : ~astropy.units.C, optional
-            Charge on the body
-        R : ~astropy.units
-            Radius of the body
-        differential : ~einsteinpy.coordinates.differential.*, optional
-            Complete coordinates of the body
-        parent : Body, optional
-            The parent object of the body
-            Useful in case of multibody systems
-        """
-        if differential:
-            if differential.system == "Cartesian":
-                self.pos_vec = [differential.x, differential.y, differential.z]
-                self.vel_vec = [differential.v_x, differential.v_y, differential.v_z]
+    name: str = "Generic Body"
+    mass: u.kg = 0 * u.kg
+    q: u.C = 0 * u.C
+    R: u.km = 0 * u.km
+    coords = None
+    parent: "Body" = None
+
+    def __post_init__(self):
+        diff = self.differential
+        if diff:
+            if diff.system == "Cartesian":
+                self.pos_vec = [
+                    diff.x,
+                    diff.y,
+                    diff.z,
+                ]
+                self.vel_vec = [
+                    diff.v_x,
+                    diff.v_y,
+                    diff.v_z,
+                ]
             else:
-                self.pos_vec = [differential.r, differential.theta, differential.phi]
-                self.vel_vec = [differential.v_r, differential.v_th, differential.v_p]
-        self.name = name
-        self.mass = mass
-        self.q = q
-        self.R = R
-        self.coords = differential
-        self.parent = parent
+                self.pos_vec = [
+                    diff.r,
+                    diff.theta,
+                    diff.phi,
+                ]
+                self.vel_vec = [
+                    diff.v_r,
+                    diff.v_th,
+                    diff.v_p,
+                ]
 

```

Comment:
Since `@u.quantity_input(mass=u.kg, q=u.C, R=u.km)` is no longer available to `__init__` to enforce type (physical unit) checks, add unit checks in the `__post_init__` method. Something like this:
```py
@u.quantity_input(mass=u.kg, q=u.C, R=u.km)
def check_units(mass, q, R):
    return mass, q, R
check_units(self.mass, self.q, self.R)
```
Keep this at the top of the method.